# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/ipc/WebrtcTCPSocket.cpp
# Commit: 8196045d38b5
# Full Hash: 8196045d38b50e04313725215c36ba1618c50486
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Create an ipdl struct called WebrtcProxyConfig, to cut down on the arg count on lots of functions. r=mjf,mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45562
# ==============================================================================

diff -r 6e1eba5c88ad -r 8196045d38b5 media/mtransport/ipc/WebrtcTCPSocket.cpp
--- a/media/mtransport/ipc/WebrtcTCPSocket.cpp	Wed Sep 18 14:27:42 2019 +0000
+++ b/media/mtransport/ipc/WebrtcTCPSocket.cpp	Tue Sep 17 18:19:04 2019 +0000
@@ -139,9 +139,8 @@
 
 nsresult WebrtcTCPSocket::Open(
     const nsCString& aHost, const int& aPort, const nsCString& aLocalAddress,
-    const int& aLocalPort, bool aUseTls, const Maybe<net::LoadInfoArgs>& aArgs,
-    const Maybe<nsCString>& aAlpn,
-    const Maybe<NrSocketProxyConfig::ProxyPolicy>& aProxyPolicy) {
+    const int& aLocalPort, bool aUseTls,
+    const Maybe<net::WebrtcProxyConfig>& aProxyConfig) {
   LOG(("WebrtcTCPSocket::Open %p\n", this));
 
   if (NS_WARN_IF(mOpened)) {
@@ -168,15 +167,9 @@
   mTls = aUseTls;
   mLocalAddress = aLocalAddress;
   mLocalPort = aLocalPort;
+  mProxyConfig = aProxyConfig;
 
-  if (aProxyPolicy.isSome() &&
-      *aProxyPolicy != NrSocketProxyConfig::kDisableProxy) {
-    mLoadInfoArgs = *aArgs;
-    mAlpn = *aAlpn;
-    if (*aProxyPolicy == NrSocketProxyConfig::kForceProxy) {
-      mForceProxy = true;
-    }
-
+  if (mProxyConfig.isSome()) {
     // We need to figure out whether a proxy needs to be used for mURI before
     // we can start on establishing a connection.
     rv = DoProxyConfigLookup();
@@ -256,7 +249,8 @@
 }
 
 nsresult WebrtcTCPSocket::OpenWithoutHttpProxy(nsIProxyInfo* aSocksProxyInfo) {
-  if (NS_WARN_IF(mForceProxy && !aSocksProxyInfo)) {
+  if (NS_WARN_IF(mProxyConfig.isSome() && mProxyConfig->forceProxy() &&
+                 !aSocksProxyInfo)) {
     return NS_ERROR_FAILURE;
   }
 
@@ -340,7 +334,7 @@
   }
 
   nsCOMPtr<nsILoadInfo> loadInfo;
-  Maybe<net::LoadInfoArgs> loadInfoArgs = Some(mLoadInfoArgs);
+  Maybe<net::LoadInfoArgs> loadInfoArgs = Some(mProxyConfig->loadInfoArgs());
   rv = LoadInfoArgsToLoadInfo(loadInfoArgs, getter_AddRefs(loadInfo));
   if (NS_FAILED(rv)) {
     LOG(("WebrtcTCPSocket %p: could not init load info\n", this));
@@ -389,7 +383,7 @@
     return NS_ERROR_FAILURE;
   }
 
-  rv = httpChannel->HTTPUpgrade(mAlpn, this);
+  rv = httpChannel->HTTPUpgrade(mProxyConfig->alpn(), this);
   if (NS_WARN_IF(NS_FAILED(rv))) {
     return rv;
   }